hphp/system/php/spl/datastructures/SplFixedArray.php

```diff
@@ -245,16 +245,21 @@ public function toArray() {
    */
   public static function fromArray($array, $save_indexes = true) {
     $fixed_array = new self;
+    if (!is_array($array)) {
+      trigger_error(
+        sprintf("SplFixedArray::fromArray() expects parameter 1 to be array,".
+                " %s given", gettype($array)),
+        E_WARNING);
+      return $fixed_array;
+    }
     if ($save_indexes) {
+      $fixed_array->setSize(max(array_keys($array)) + 1);
       foreach ($array as $key => $value) {
         if (!is_numeric($key) || $key < 0) {
           throw new InvalidArgumentException(
             'array must contain only positive integer keys'
           );
         }
-        if ($key >= $fixed_array->count()) {
-          $fixed_array->setSize($key + 1);
-        }
         $fixed_array[$key] = $value;
       }
```

Comment:
If somebody is wondering why we don't just reuse `$array` and only do a validation loop, the answer is (probably) that we need to break references, so we have to do a hard copy.

It's not related to your change, but that `is_numeric` check should be `is_int`. We don't want `"1.2"` as a key in a fixed array.